builtin/credential/approle/path_login.go

```diff
@@ -389,16 +389,29 @@ func (b *backend) pathLoginUpdate(ctx context.Context, req *logical.Request, dat
 
 // Invoked when the token issued by this backend is attempting a renewal.
 func (b *backend) pathLoginRenew(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	roleName := req.Auth.InternalData["role_name"].(string)
-	if roleName == "" {
+	// Fetch the role_name from internal data
+	roleName, ok := req.Auth.InternalData["role_name"].(string)
+	if !ok || roleName == "" {
```

Comment:
Given `InternalData` is a `map[string]string`, I think this is equivalent to the [existing logic](https://go.dev/play/p/AemIH7kDzt5)?